### PR TITLE
fragment caching, partial as first argument

### PR DESCRIPTION
**Pull request by skatkov**
Adds ability to provide partial as first argument.

```ruby
pb.friends "racers/racer", as: :racer, collection: @racers
```

Also in detail explains how fragment caching works now and tests that verify that it stays working.

### CODE REVIEW

**Review by svanhesteren — COMMENTED** [review]
Code looks good I think, but when facing the www I think we can do a bit better in the use of english in the comments. This will look a bit more professional. Hope you agree with me. 🙂

File: lib/pbbuilder/template.rb [hunk 1]
```diff
@@ -80,7 +59,12 @@ def set!(field, *args, **kwargs, &block)
         end
       end
     else
-      super
+      if args.one? && kwargs.has_key?(:collection) && kwargs.has_key?(:as)
+        # pb.friends "racers/racer", as: :racer, collection: [Racer.new(1, "Johnny Test", []), Racer.new(2, "Max Verstappen", [])]
```
[the remark this answers is not in the document]
Comment (skatkov) [reply lib/pbbuilder/template.rb:63]: I found it very useful to have example syntax that will be handle by this branch of code. I'll annotate thanks.

**Comment by skatkov** [comment]
@svanhesteren thanks for a review! Adjusted PR according to your feedback

File: lib/pbbuilder/template.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (svanhesteren) [reply lib/pbbuilder/template.rb]: Ah ok. I wasn't sure if it was a leftover or not. :)

**Review by svanhesteren — APPROVED** [review]

**Pull request merged**